diffsync/__init__.py

```diff
@@ -329,7 +329,7 @@ def add_child(self, child: "DiffSyncModel"):
         attr_name = self._children[child_type]
         childs = getattr(self, attr_name)
         if child.get_unique_id() in childs:
-            raise ObjectAlreadyExists(f"Already storing a {child_type} with unique_id {child.get_unique_id()}")
+            raise ObjectAlreadyExists(f"Already storing a {child_type} with unique_id {child.get_unique_id()}", child)
```

Comment:
It'd be nice to change the `__init__` method of `ObjectAlreadyExists` to make the "existing object" arg mandatory and cache it in a more useful location than `self.value.args[1]`, something like:

```python
class ObjectAlreadyExists(ObjectStoreException):
    """Exception raised when trying to store a DiffSyncModel or DiffElement that is already being stored."""

    def __init__(self, message, existing_object, *args, **kwargs):
        self.existing_object = existing_object
        super().__init__(message, existing_object, *args, **kwargs)
```

Then your tests can check for `error.existing_object` instead of `error.value.args[1]`.

Reply:
Ok I updated it to do this, but we do have to access it via `error.value.existing_object`.